File: nodejs/programmers/Greedy/42862.js
function solution(n, lost, reserve) {
    n -= lost.length;
    let _lost = [];

    lost.forEach((lst) => _lost.push(lst));
    _lost.forEach(function (lst){
        if(reserve.includes(lst)){
            reserve.splice(reserve.indexOf(lst),1);
            lost.splice(lost.indexOf(lst),1);
            n+=1;
        }
    })

    lost.forEach(function(lst){
        let removed;
        if(reserve.includes(lst)) removed = lst;
        else if(reserve.includes(lst-1)) removed = lst-1;
        else if(reserve.includes(lst+1)) removed = lst+1;
        if(removed){
            reserve.splice(reserve.indexOf(removed),1);
            n+=1;
        }
    })
    return n;
}

console.log(
    solution(5, [1, 2, 3], [2, 3, 4])
);

File: programmers/level1/42862.cpp
#include <bits/stdc++.h>

using namespace std;

int solution(int n, vector<int> lost, vector<int> reserve) {
    vector<int> v(n + 1, 1); v[0] = 0;
    for (auto _lst : lost   ) v[_lst] = 0;
    for (auto _rsv : reserve) v[_rsv] += 1;
    for (int i = 1; i <= n; ++i) {
        if (v[i] == 0) {
            if (v[i - 1] == 2) {
                v[i - 1] = 1;
                v[i] = 1;
                continue;
            }
            else if (i != n && v[i + 1] == 2) {
                v[i + 1] = 1;
                v[i] = 1;
                continue;
            }
        }
    }
    int rst = 0;
    for (auto val : v) if (val) rst++;
    return rst;
}